File: frontend/src/services/descuento.service.ts
import { apiClient } from '@/lib/api-client'
import type { DescuentoDTORequest, DescuentoDTOResponse } from '@/types'

export const descuentoService = {
  async listarDescuentos(idEspacioTrabajo: string): Promise<DescuentoDTOResponse[]> {
    const { data } = await apiClient.get<DescuentoDTOResponse[]>(
      `/descuento/espacio/${idEspacioTrabajo}`
    )
    return data
  },

  async crearDescuento(descuento: DescuentoDTORequest): Promise<void> {
    await apiClient.post('/descuento', descuento)
  },

  async eliminarDescuento(id: number): Promise<void> {
    await apiClient.delete(`/descuento/${id}`)
  },
}

File: backend/src/main/java/com/campito/backend/descuentos/service/DescuentoService.java
package com.campito.backend.descuentos.service;

import java.util.List;
import java.util.UUID;

import com.campito.backend.descuentos.domain.dto.DescuentoDTORequest;
import com.campito.backend.descuentos.domain.dto.DescuentoDTOResponse;

public interface DescuentoService {
    public DescuentoDTOResponse crearDescuento(DescuentoDTORequest dto);
    public List<DescuentoDTOResponse> listarDescuentos(UUID idEspacioTrabajo);
    public void eliminarDescuento(Long id);
}
